### ISSUE TITLE
Gatsby unable to recognize gatsby-source-sanity plugin

### ISSUE THREAD

**Issue by johnbarrierwilson** [sanity-io/gatsby-source-sanity#20]
Hey there! 👋

I recently created a Sanity project with some basic document types. All document types also have at least placeholder values included (i.e. nothing is "empty" in the Sanity project).

After setting up the Sanity project, I then began a new Gatsby site by running `gatsby new`. Once everything was installed and set up, I immediately ran `yarn add gatsby-source-sanity` and started setting up the `gatsby-config.js` file.

Here's what my `gatsby-config.js` looks like at this point:

```
require('dotenv').config()

module.exports = {
  siteMetadata: {
    title: `Gatsby Default Starter`,
    description: `Kick off your next, great Gatsby project with this default starter. This barebones starter ships with the main Gatsby configuration files you might need.`,
    author: `@gatsbyjs`,
  },
  plugins: [
    {
      resolve: 'gatsby-source-sanity',
      options: {
        projectId: '7gnb8e5o',
        dataset: 'production',
        token: process.env.SANITY_TOKEN,
        watchMode: true,
        overlayDrafts: true
      }
    },
    `gatsby-plugin-react-helmet`,
    {
      resolve: `gatsby-source-filesystem`,
      options: {
        name: `images`,
        path: `${__dirname}/src/images`,
      },
    },
    `gatsby-transformer-sharp`,
    `gatsby-plugin-sharp`,
    {
      resolve: `gatsby-plugin-manifest`,
      options: {
        name: `gatsby-starter-default`,
        short_name: `starter`,
        start_url: `/`,
        background_color: `#663399`,
        theme_color: `#663399`,
        display: `minimal-ui`,
        icon: `src/images/gatsby-icon.png`,
      },
    }
  ],
}
```

After that, I ran `yarn develop` to begin the Gatsby development server. Everything builds properly and the site loads, except that it does not allow me to query anything Sanity-related in the the Graph*i*QL explorer, nor can it query anything Sanity-related in the codebase itself.

Versions:
```
"gatsby": "^2.3.5",
"gatsby-image": "^2.0.37",
"gatsby-plugin-manifest": "^2.0.26",
"gatsby-plugin-offline": "^2.0.25",
"gatsby-plugin-react-helmet": "^3.0.11",
"gatsby-plugin-sharp": "^2.0.32",
"gatsby-source-filesystem": "^2.0.28",
"gatsby-source-sanity": "^2.0.3",
"gatsby-transformer-sharp": "^2.1.17",
"prop-types": "^15.7.2",
"react": "^16.8.6",
"react-dom": "^16.8.6",
"react-helmet": "^5.2.0"
```

**In summary**:

1. Set up a basic Sanity project (with `sanity init`)
2. Set up basic Gatsby project (with `gatsby new`)
3. Added `gatsby-source-sanity`
4. Set up `gatsby-config.js` with the new resolver (see above)
5. Ran `yarn develop`
6. No luck being able to create Sanity-related queries.

**Comment by johnbarrierwilson** [comment]
There's been some discussion here, but I still had no luck even after running `gatsby clean`: https://twitter.com/jlengstorf/status/1113913762585518080

**Comment by rexxars** [comment]
Seems Gatsby introduced a bug recently that silently ignores the plugin, but it was fixed in gatsby@2.3.10

**Comment by johnbarrierwilson** [comment]
Ah ha! That did it. Thanks @rexxars!
